src/lib/blkdata_svc/blkdata_service.cpp

```diff
@@ -212,7 +212,10 @@ folly::Future< std::error_code > BlkDataService::async_free_blk(MultiBlkId const
     auto f = promise.getFuture();
 
     m_blk_read_tracker->wait_on(bids, [this, bids, p = std::move(promise)]() mutable {
-        m_vdev->free_blk(bids);
+        {
+            auto cpg = hs()->cp_mgr().cp_guard();
```

Comment:
We only take cp_guard for bitmapblk allocator, for appendblkallocator, it would just call m_vdev->free_blk(), right?

Reply:
I didn't want to do a specific call for different blk allocators. Taking CPGuard is an extremely light weight operation and IMO we don't want to optimize, rather like to have clean code.

Reply:
Taking additional cpguard is fine. 
With this change, free_blk won't take effect in next cp_flush for appendblkallocator, but for next cp_flush, since appendblkallocator::free() will take cpgurad under vdev::cp_flush() (which should be fine, right?), and it will be flushed until next cp, which is not a big deal.

But it seems accumulating the free blk list is unnecessary for appendblkallocator, which seems a little odd.

Reply:
I see, yes we don't want to accumulate that for append_blk_allocator. But without adding additional parameters such as do_instant_free, we can't do that, let me see how to overcome this. Thanks!

Reply:
How about let vdev::free_blk take cpguard and differentiate whether to accumulate or not based on allocator type?